src/components/documents-store.ts

```diff
@@ -0,0 +1,84 @@
+import { Document, IDocumentsStore, OpaqueDocumentStore, OpaqueIndex, Schema } from "../types.js";
+
+export interface DocumentsStore extends OpaqueDocumentStore {
+  docs: Record<string, Document | undefined>;
+  count: number;
+}
+
+type DefaultDocumentsStore<S extends Schema, I extends OpaqueIndex> = IDocumentsStore<S, I, DocumentsStore>;
+
+function create(): DocumentsStore {
+  return {
+    docs: {},
+    count: 0,
+  };
+}
+
+function get(store: DocumentsStore, id: string): Document | undefined {
+  return store.docs[id];
+}
+
+function getMultiple(store: DocumentsStore, ids: string[]): (Document | undefined)[] {
+  const found: (Document | undefined)[] = Array.from({ length: ids.length });
+
+  for (let i = 0; i < ids.length; i++) {
+    found[i] = store.docs[ids[i]];
+  }
+
+  return found;
+}
+
+function store(store: DocumentsStore, id: string, doc: Document): boolean {
+  if (typeof store.docs[id] !== "undefined") {
```

Comment:
This store is currently internal. At the moment I just copied the existing implementation, but we can change it later to allow that.
The only point I have is that it's a "all or nothing" thing. The parameter must be provided when creating the store and it will apply to ALL operations.

Reply:
got it!